contracts/.github/workflows/test.yml

```diff
@@ -13,22 +13,59 @@ jobs:
     name: Foundry project
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout repository
+        uses: actions/checkout@v4
         with:
           submodules: recursive
+        
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12.x
+          check-latest: true
+          cache: pip
 
       - name: Install Foundry
         uses: foundry-rs/foundry-toolchain@v1
         with:
           version: nightly
+          cache: true
 
-      - name: Run Forge build
+      - name: Install Bun
+        uses: oven-sh/setup-bun@v2
+        with:
+          bun-version: latest
+      
+      - name: Install Python dependencies
+        run: |
+          pip install -U setuptools
+          pip install -r requirements.txt
+
+      - name: Build code
         run: |
           forge --version
           forge build --sizes
         id: build
+      
+      - name: Check file and code format
+        if: false
+        run: bun run format
```

Comment:
since you already intoduced the package.json with bunch of script calls i would advise to call all the commands through that package.json (npm) so we have consistent way to reproduce some failures on ci :thinking:

Reply:
bun is a lightweight Node.js like runtime, which faster and consumes less resources in comparison Node.js https://bun.sh/. As forge is not a Typescript-dependent framework, bun will do. Moreover it's used in the most popular forge template https://github.com/PaulRBerg/foundry-template/tree/main.